contracts/strategies/StrategyEthHegicLP.sol

```diff
@@ -140,17 +132,25 @@ contract StrategyEthHegicLP is BaseStrategy {
     {
         // Shouldn't we revert if we try to exitPosition and there is a timelock?
 
-        uint256 writeEth = IERC20(ethPool).balanceOf(address(this));
-        uint256 _timeLock = withdrawLockRemaining();
+        uint256 writeEth = IHegicEthPool(ethPool).shareOf(address(this));
+        uint256 stakingBalance = IHegicEthPoolStaking(ethPoolStaking).balanceOf(address(this));
 
-        // timelock will never be negative
-        if (_timeLock <= 0) {
-            uint256 writeBurn = writeEth.add(1);
+        // by doing this before the timelock check, we will trigger the timelock
+        if (stakingBalance > 0) {
             IHegicEthPoolStaking(ethPoolStaking).exit();
+        }
+
+        // timelock will never be negative now that we've changed it to boolean
+        bool unlocked = withdrawUnlocked();
+        if (unlocked = true) {
```

Comment:
this is a common mistake. Some c developers usually do `true == variable` to avoid if statements with assignments returning true all the time :)

Reply:
thanks, fixed!